bibtex-actions.el

```diff
@@ -225,53 +244,75 @@ offering the selection candidates"
+                         return (cdr (assoc-string fname item 'case-fold)))
+      ""))
+
+ (defun bibtex-actions-shorten-names (names)
+  "Return a list of family names from a list of full NAMES.
+
+To better accomomodate corporate names, this will only shorten
+personal names of the form 'family, given'."
+  (mapconcat
+   (lambda (name)
+     (if (eq 1 (length name))
+         (cdr (split-string name " "))
+       (car (split-string name ", "))))
+   (split-string names " and ") ", "))
+
 (defun bibtex-actions--format-candidates (&optional context)
   "Format candidates, with optional hidden CONTEXT metadata.
 This both propertizes the candidates for display, and grabs the
 key associated with each one."
-  (let* ((main-template
+  (let* ((candidates
+          (ht-values (parsebib-parse bibtex-completion-bibliography)))
+         (main-template
          (bibtex-actions--process-display-formats
           bibtex-actions-template))
          (suffix-template
           (bibtex-actions--process-display-formats
            bibtex-actions-template-suffix))
          (main-width (truncate (* (frame-width) 0.65)))
          (suffix-width (truncate (* (frame-width) 0.34))))
-    (cl-loop
-     for candidate in (bibtex-completion-candidates)
-     collect
-     (let* ((pdf (if (assoc "=has-pdf=" (cdr candidate)) " has:pdf"))
-            (note (if (assoc "=has-note=" (cdr candidate)) "has:note"))
-            (link (if (or (assoc "doi" (cdr candidate))
-                          (assoc "url" (cdr candidate))) "has:link"))
-            (citekey (bibtex-completion-get-value "=key=" candidate))
-            (candidate-main
-             (bibtex-actions--format-entry
-              candidate
-              main-width
-              main-template))
-            (candidate-suffix
-             (bibtex-actions--format-entry
-              candidate
-              suffix-width
-              suffix-template))
-            ;; We display this content already using symbols; here we add back
-            ;; text to allow it to be searched, and citekey to ensure uniqueness
-            ;; of the candidate.
-            (candidate-hidden (s-join " " (list pdf note link context citekey))))
-       (cons
-        ;; If we don't trim the trailing whitespace, 'completing-read-multiple' will
-        ;; get confused when there are multiple selected candidates.
-        (s-trim-right
-         (concat
-          ;; We need all of these searchable:
-          ;;   1. the 'candidate-main' variable to be displayed
-          ;;   2. the 'candidate-suffix' variable to be displayed with a different face
-          ;;   3. the 'candidate-hidden' variable to be hidden
-          (propertize candidate-main 'face 'bibtex-actions-highlight) " "
-          (propertize candidate-suffix 'face 'bibtex-actions) " "
-          (propertize candidate-hidden 'invisible t)))
-        citekey)))))
+    (cl-loop for candidate in candidates
```

Comment:
`cl-loop` can loop over the values of a hash table directly, I think this should work
```suggestion
    (cl-loop for candidate being the hash-values of (parsebib-parse bibtex-completion-bibliography)
```
and will allow you to get rid of dependency on `ht`.

Reply:
So your suggestion is two-fold:

1. add a required arg on the function
2. do the suggestion here, but instead reference that arg?

BTW, have you had a chance to test this branch?

Reply:
I was also wondering if I need to add a `bibtex-actions-bibliography` defcustom to hold that list?

Reply:
I just pushed a change with these three things.

Still have to test, but WDYT?

Reply:
> So your suggestion is two-fold:
> 
>     1. add a required arg on the function
> 
>     2. do the suggestion here, but instead reference that arg?

Yes.

> BTW, have you had a chance to test this branch?

No, I will wait till filenotify stuff is merged since I will need to change my config without it.

Reply:
> Still have to test, but WDYT?

Actually, don't think it works as is, as the `ht-values` function returns the values for all the keys.

But should be easy to fix.

Reply:
> Nevermind; missed something :-)
> 
> Works now.

Local cache still needs to be adjusted. I added a suggestion (hopefully with balanced parens).

Reply:
> I was also wondering if I need to add a `bibtex-actions-bibliography` defcustom to hold that list?

What is the eventual goal? Removal of dependency on `bibtex-completion`? It seems worthwhile though the commands will need to be re-implemented.

Maybe for the time being it should be defaulted `bibtex-completions-bibliography` otherwise it will break some people's setup

Reply:
> > I was also wondering if I need to add a `bibtex-actions-bibliography` defcustom to hold that list?
> 
> What is the eventual goal? Removal of dependency on `bibtex-completion`? 

Yes.

> It seems worthwhile though the commands will need to be re-implemented.
> 
> Maybe for the time being it should be defaulted `bibtex-completions-bibliography` otherwise it will break some people's setup

Right.

I guess I should really remove the defcustom for now then?

Reply:
... or just keep it but set to default to bibtex-completion-bibliography?

Reply:
> ... or just keep it but set to default to bibtex-completion-bibliography?

I think you can use `bibtex-completion` as an additional choice and the default value in the defcustom, set it to the default and maybe say it will be removed later.